### PR TITLE
Remove phase referencing in RotationAndDiagonalConstraint

### PR DESCRIPTION
**Pull request by tammojan**
The phase referencing here is not necessary and will fail if the
first station is flagged.

### CODE REVIEW

**Comment by aroffringa** [comment]
Don't you want to make it optional first, through a DDECal parset option (and maybe disable it by default) ?

**Comment by rvweeren** [comment]
I think disable by default is preferred, like we did for TEC (having a DDECal parset option might be useful to verify the behavior is correct).

**Comment by tammojan** [comment]
Optional is an option yes. It'll take me some time to propagate the option here...

**Comment by tammojan** [comment]
Yes I'll have to make it optional if only for the tests, otherwise there is no deterministic outcome for the test (which now [fails](https://travis-ci.org/github/lofar-astron/DP3/jobs/691667224#L12001)):
```
/8 Test #8: rotationconstraint.test_rotation_and_diagonal ...***Failed    0.05 sec

Running 1 test case...

/src/DDECal/test/tRotationConstraint.cc(76): error: in "rotationconstraint/test_rotation_and_diagonal": check near(constraint_result[0].vals[0], 0.) has failed

*** 1 failure is detected in the test module "DP3"
```

**Comment by tammojan** [comment]
Argh, I was confused about phase referencing. It's actually about rotation referencing here..

File: DDECal/test/tRotationConstraint.cc [hunk 1]
```diff
@@ -48,11 +49,13 @@ BOOST_AUTO_TEST_CASE( test_rotation )
...
 
-BOOST_AUTO_TEST_CASE( test_rotation_and_diagonal )
+BOOST_DATA_TEST_CASE( test_rotation_and_diagonal,
+                      boost::unit_test::data::make({ true, false }),
+                      doPhaseReference )
 {
   RotationAndDiagonalConstraint constraint;
   constraint.InitializeDimensions(1, 1, 1);
-  constraint.SetDoPhaseReference(true);
+  constraint.SetDoPhaseReference(doPhaseReference);
```
Comment (tammojan) [inline DDECal/test/tRotationConstraint.cc:58]: So sorry, this should have been `doRotationReference`...

File: DDECal/test/tRotationConstraint.cc [hunk 2]
```diff
@@ -48,11 +49,13 @@ BOOST_AUTO_TEST_CASE( test_rotation )
...
 
   vector<vector<complex<double> > > onesolution(1);
```
Comment (tammojan) [inline DDECal/test/tRotationConstraint.cc:60]: To test the constraint, you'll need at least two solutions (this dimension is the number of antennas).

File: DDECal/test/tRotationConstraint.cc (hunk 1 above)
Comment (mnijhuis-tos) [reply DDECal/test/tRotationConstraint.cc]: Should I also update the parset key in DDECal.cc?

Comment (tammojan) [reply DDECal/test/tRotationConstraint.cc]: Yes, but only for this particular constraint; for the existing constraints (tec etc) it should stay phasereference.

Comment (tammojan) [reply DDECal/test/tRotationConstraint.cc]: So you should add a parset key `rotationreference`.

File: DDECal/DDECal.cc [hunk 3]
```diff
@@ -291,9 +281,13 @@ void DDECal::initializeConstraints(const ParameterSet& parset, const string& pre
 #endif
       break;
     case GainCal::ROTATIONANDDIAGONAL:
-      itsConstraints.emplace_back(new RotationAndDiagonalConstraint());
+    {
+      auto constraintPtr = boost::make_unique<RotationAndDiagonalConstraint>();
+      constraintPtr->SetDoRotationReference(parset.getBool(prefix + "rotationreference", true));
```
Comment (tammojan) [inline DDECal/DDECal.cc:286]: as per discussion above, the default should be false

File: docs/schemas/DDECal.yml [hunk 4]
```diff
@@ -141,3 +141,17 @@ inputs:
   applycal&#46;*:
     type: 
     doc: ApplyCal sub-step, same as in Predict step. One can pass an h5parm with as many directions as set in "directions" and each direction model is corrupted accordingly `.`
+  phasereference:
+    default: true
+    type: bool
+    doc: Use phase referencing. This setting is only effective when mode=tec or mode=tecandphase `.`
```
Comment (tammojan) [inline docs/schemas/DDECal.yml:147]: Could you explain the need for the backtick here? I think it's prettier when `mode=tec` is rendered as code.

Comment (mnijhuis-tos) [reply docs/schemas/DDECal.yml]: The tools we use for generating the html pages require the `.` at the end. I've prettyfied the mode=... texts.

**Pull request merged**